src/kbucket.rs

```diff
@@ -1,219 +1,87 @@
-            }
-            Search::Failure(bucket_index, column_index) => {
-                let bucket = self.buckets[bucket_index];
-                let mut known_nodes = Vec::new();
-
-                for node in bucket.iter() {
-                    if node.is_some() {
-                        // Should I be dereferencing the node to send to others?  Or copy the node to share?
-                        known_nodes.push(*node)
-                    }
-                }
-                FindNodeResult::NotFound(known_nodes)
-            }
+            local_record,
+            buckets: vec![Default::default(); MAX_BUCKETS],
         }
     }
-    // TODO:
-    pub fn find_value() {}
 
-    // TODO:
-    /// Instructs a node to store a key, value pair for later retrieval.
-    ///
-    /// "Most operations are implemented in terms of the lookup proceedure. To store a
-    /// <key,value> pair, a participant locates the k closes nodes to the key and sends them store RPCs".
-    pub fn store(&mut self, key: Identifier, value: Vec<u8>) {}
-
-    // Non-RPCs:
-    // ---------------------------------------------------------------------------------------------------
-    fn add_node(&mut self, node: &Node) -> bool {
-        match self.search_table(node.node_id) {
-            Search::Success(bucket_index, column_index) => false,
-            Search::Failure(bucket_index, column_index) => {
-                self.buckets[bucket_index][column_index] = Some(*node);
-                true
-            }
+    // TODO: Remove bool return statement (used in tests rn)
+    pub fn add(&mut self, record: TableRecord) -> bool {
+        let bucket_index = self.xor_bucket_index(&record.node_id);
+        match self.buckets[bucket_index].insert(record.node_id, record) {
+            Some(_) => false,
+            None => true,
         }
     }
 
-    fn search_table(&self, id: Identifier) -> Search {
+    pub fn get(&self, id: &Identifier) -> Option<TableRecord> {
         let mut last_empty_index = 0;
-        let bucket_index = self.xor_bucket_index(id);
-        let mut bucket = self.buckets[bucket_index];
+        let bucket_index = self.xor_bucket_index(&id);
+        let mut bucket = self.buckets[bucket_index].clone();
 
-        for (i, node) in bucket.iter().enumerate() {
-            match node {
-                Some(bucket_node) => {
-                    if bucket_node.node_id == id {
-                        return Search::Success(bucket_index, i);
-                    } else {
-                        continue;
-                    };
-                }
-                _ => {
-                    last_empty_index = i;
-                }
+        for (i, node) in bucket.map.iter().enumerate() {
+            if node.0 == id {
+                let record = node.1.clone();
+                return Some(record);
             }
         }
-        Search::Failure(bucket_index, last_empty_index)
+        None
```

Comment:
you can just use `get` https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.get here and avoid even the `mut` on the bucket above